File: scripts/list-assets.ts
import * as https from 'https';

interface GithubContentItem {
  name: string;
  [key: string]: unknown;
}

const options = { headers: { 'User-Agent': 'node' } };
https.get(
  'https://api.github.com/repos/fghrsh/live2d_demo/contents/assets',
  options,
  (res) => {
    let body = '';
    res.on('data', (c: string) => (body += c));
    res.on('end', () => {
      const js: GithubContentItem[] = JSON.parse(body);
      console.log(js.map((i) => i.name));
    });
  }
);

File: scripts/list-assets.py
import requests

url = 'https://api.github.com/repos/fghrsh/live2d_demo/git/trees/master?recursive=1'
headers = {'User-Agent': 'py'}
resp = requests.get(url, headers=headers)
resp.raise_for_status()
js = resp.json()
results = [item['path'] for item in js.get('tree', []) if 'koharu' in item['path'] and item['path'].endswith('.json')]
print(results)
